tests/unit/regression/lars_unittest.cc

```diff
@@ -233,6 +233,48 @@ TEST(LeastAngleRegression, lars_N_LessThan_D)
 	SG_UNREF(labels);
 }
 
+TEST(LeastAngleRegression, lars_n_less_than_d_32_feature_test)
+{
+	SGMatrix<float64_t> data_64(5,3);
+	SGVector<float64_t> lab(3);
+	generate_data_n_less_d(data_64,lab);
+
+	//copy data_64 into a float32_t SGMatrix
+	SGMatrix<float32_t> data(5,3);
+	for(index_t c = 0; c < data_64.num_cols; ++c)
+		for(index_t r = 0; r < data_64.num_rows; ++r)
+			data(r, c) = (float32_t) data_64(r, c);
+
+
+	CDenseFeatures<float32_t>* features = new CDenseFeatures<float32_t>(data);
```

Comment:
or we could even have a template for unit tests that just exchanges the features in train and results.
Like automating the test that @c4goldsw wrote here based on the 64 bit version